stable_plugins/visualization/file_types/qasm_visualization_templates/qasm_visualization.html

```diff
@@ -0,0 +1,102 @@
+{% import 'forms.html' as forms %}
+
+<html>
+
+<head>
+    <link rel="stylesheet" type="text/css" href="{{url_for('static', filename='forms.css')}}" media="screen" />
+    <style>
+        .qhana-data-preview {
+            text-rendering: optimizeSpeed;
+            display: flex;
+            flex-direction: row;
+            flex-wrap: wrap;
+            max-width: 100%;
+            max-height: 750px;
+            overflow: auto;
+            background: transparent;
+        }
+
+        #spinner {
+            display: inline-block;
+            width: 100px;
+            height: 100px;
+            border: 6px solid rgba(0, 0, 0, 0);
+            border-radius: 50%;
+            border-top-color: #673ab7;
+            animation: spin 1s linear infinite;
+            -webkit-animation: spin 1s linear infinite;
+        }
+
+        .container {
+            display: flex;
+            justify-content: center;
+            align-items: center;
+            width: 100%;
+        }
+
+        @keyframes spin {
+            to {
+                -webkit-transform: rotate(360deg);
+            }
+        }
+    </style>
+</head>
+
+<body>
+    <div class="qhana-microfrontend">
+        <details id="visualization-options" open>
+            <summary>Visualization Options</summary>
+            {% call forms.render_form(target="microfrontend") %}
+            {{ forms.render_fields(schema, values=values, errors=errors) }}
+            <div class="qhana-form-buttons">
+                {{ forms.submit("validate", target="microfrontend") }}
+            </div>
+            {% endcall %}
+        </details>
+    </div>
+
+    <img id="circuit-image" class="qhana-data-preview" onload="onCircuitImageLoad()" onerror="onCircuitImageError()">
+    <div class="container">
+        <div id="spinner"></div>
+    </div>
+
+    <script>
+        let currentUrl = null;
+
+        function getUrl() {
+            return document.querySelector('input[name=data]').value;
+        }
+
+        function updatePreview() {
+            const circuitUrl = getUrl()
+            if (currentUrl !== circuitUrl) {
+                document.querySelector('#spinner').style.display = '';
+                const preview = document.querySelector('#circuit-image');
+                const url = "{{ url_for(request.endpoint)[:-3] }}circuits?circuit-url=" + circuitUrl;
```

Comment:
You can just calculate the correct URL in the python code and pass it as a context var or hardcode the correct endpoint name in the template. Either way, do not string concat together parts for the url but pass everything (including query params) to the `url_for` function:
`url_for('<blueprint-name>.<endpoint-name>`, **{'circuit-url': circuitUrl})`

(Dictionary unpacking `**` is required for url params with `-`)
The correct endpoint name can be found using the `/debug/routes` endpoint that is available in the dev setup.